collect_app/src/main/java/org/odk/collect/android/upload/InstanceGoogleSheetsUploader.java

```diff
@@ -294,8 +296,14 @@ private String uploadMediaFile(Instance instance, String fileName) throws Upload
 
     private TreeElement getInstanceElement(String formFilePath, File instanceFile) throws UploadException {
         FormDef formDef;
+
+        // Get the last-saved instance, if it exists.
+        File formXml = new File(formFilePath);
+        File lastSavedFile = FileUtils.getLastSavedFile(formXml);
+        String lastSavedSrc = lastSavedFile.exists() ? "jr://file/" + LAST_SAVED_FILENAME : null;
```

Comment:
It's a little hard to see that this URL points to the file at `lastSavedFile` (you have to follow `LAST_SAVED_FILENAME` to `FormController` and then to `FileUtils` and then to `FileUtils.getLastSavedFile`).  How about a `FileUtils.getLastSavedUrl` that returns a nullable String?

Reply:
Ha x 2, I see that you already did this also!

Reply:
Same thing happened to me on both spots during my review :D

Reply:
@cooperka has a time machine 😄